### DISCUSSION TITLE
Odd behavior for `getAutoSortDir`

### DISCUSSION THREAD

**Discussion by colindjk** [TanStack/table#4020] [General]
Currently `getAutoSortDir` in file `table-core/src/features/Sorting.ts` has a return value dependent on the type of data received in the first row of data passed to the `useTableInstance` hook. 

I was curious, could this be updated to be consistent regardless of sort direction?

Currently I'm working with a column that can either have strings or objects as it's value, which makes the behavior of the column break since this can happen:

- Unsorted (first row is a string value)
- User triggers sort -> Sorted (Asc)
  - Asc first since row value was string
- Row value is now an object
  - Next sort order is `false` aka unsorted.

Proposal:
- Auto sort direction is always the same, unless specific differently via options.

**Comment by bboyz269** [comment]
Agree, guessing initial sort direction is really weird. Wouldn't it be better to have ascending order by default and force devs to set it?
I can't think of a case where initial sort direction of specific column requires to be dynamic.

Or at least a warning in documents / console since this odd behavior is so common with string column that has `undefined` values.
// Took me whole day to find this topic and understand what's wrong. I fixed my case of specify `sortDescFirst` for columns that might have undefined values
